client/src/pages/PasswordReset.js

```diff
@@ -0,0 +1,211 @@
+              {
+                required: true,
+                message: 'Please input your E-mail!',
+              },
+            ],
+          })(
+            <Input
+              prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
+              placeholder="E-mail"
+            />,
+          )}
+        </Form.Item>
+        <Form.Item label="*" hasFeedback className="form-text">
+          {getFieldDecorator('password', {
+            rules: [
+              {
+                required: true,
+                message: 'Please create a password!',
+              },
+            ],
+          })(
+            <Input.Password
+              prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}
+              type="password"
+              placeholder="Create Password"
+            />,
+          )}
+        </Form.Item>
+        <Form.Item label="*" hasFeedback className="form-text">
+          {getFieldDecorator('confirm', {
+            rules: [
+              {
+                required: true,
+                message: 'Please confirm your password!',
+              },
+              {
+                validator: (rule, value, password) =>
+                  compareToFirstPassword(rule, value, password),
+              },
+            ],
+          })(
+            <Input.Password
+              onBlur={e => handleConfirmBlur(e)}
+              prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}
+              type="password"
+              placeholder="Confirm Password"
+            />,
+          )}
+        </Form.Item>
+        <Form.Item name="select" label="*" hasFeedback className="form-text">
+          {getFieldDecorator('questionIdx', {
+            rules: [
+              {
+                required: true,
+                message: 'Please select your security question!',
+              },
+            ],
+          })(
+            <Select placeholder="Please select your security question!">
+              {securityQuestions.map((question, idx) => {
+                return <Option value={idx}>{question}</Option>;
+              })}
+            </Select>,
+          )}
+        </Form.Item>
+        <Form.Item label="*" hasFeedback className="form-text">
+          {getFieldDecorator('answer', {
+            rules: [
+              {
+                required: true,
+                message: 'Please type in an answer!',
+              },
+            ],
+          })(<Input placeholder="Answer" />)}
+        </Form.Item>
+        <Form.Item {...tailFormItemLayout}>
+          <Button className="reg-button" type="primary" htmlType="submit">
+            Reset
+          </Button>
+        </Form.Item>
+      </Form>
+    </div>
+  );
+};
+
+PasswordReset.propTypes = {
+  form: Form.isRequired,
+  authed: PropTypes.string.isRequired,
+  setAuthed: PropTypes.func.isRequired,
+  setAuthRole: PropTypes.func.isRequired,
+};
+
+export default Form.create()(PasswordReset);
```

Comment:
This is mostly the same as the register page, the only major differences is how that the reset password function is called instead of the register function on submit.